src/query/api/v1/handler/prometheus/native/read_instantaneous_test.go

```diff
@@ -0,0 +1,172 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package native
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/query/block"
+	"github.com/m3db/m3/src/query/test"
+	"github.com/m3db/m3/src/query/util/logging"
+	xtest "github.com/m3db/m3/src/x/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type vectorResult struct {
+	Data struct {
+		Result []struct {
+			Metric map[string]string  `json:"metric"`
+			Value  vectorResultValues `json:"value"`
+		} `json:"result"`
+	} `json:"data"`
+}
+
+type vectorResultValues []interface{}
+
+func (v vectorResultValues) parse() (time.Time, int, error) {
+	if len(v) != 2 {
+		return time.Time{}, 0,
+			fmt.Errorf("expected length 2: actual=%d", len(v))
+	}
+
+	t, ok := v[0].(float64)
+	if !ok {
+		return time.Time{}, 0,
+			fmt.Errorf("could not unmarshal time: %v", v[0])
+	}
+
+	str, ok := v[1].(string)
+	if !ok {
+		return time.Time{}, 0,
+			fmt.Errorf("could not unmarshal value: %v", v[1])
+	}
+
+	n, err := strconv.Atoi(str)
+	if err != nil {
+		return time.Time{}, 0,
+			fmt.Errorf("could not convert value to number: err=%v", err)
+	}
+
+	return time.Unix(int64(t), 0), n, nil
+}
+
+func TestPromReadInstantHandler(t *testing.T) {
+	logging.InitWithCores(nil)
+
+	values, bounds := test.GenerateValuesAndBounds(nil, nil)
+
+	setup := newTestSetup()
+	promReadInstant := setup.Handlers.InstantRead
+
+	b := test.NewBlockFromValues(bounds, values)
+	setup.Storage.SetFetchBlocksResult(block.Result{Blocks: []block.Block{b}}, nil)
+
+	req := httptest.NewRequest(PromReadInstantHTTPMethod, PromReadInstantURL, nil)
+
+	params := url.Values{}
+	params.Set(queryParam, "dummy0{}")
+
+	req.URL.RawQuery = params.Encode()
+
+	recorder := httptest.NewRecorder()
+	promReadInstant.ServeHTTP(recorder, req)
+
+	require.Equal(t, http.StatusOK, recorder.Result().StatusCode)
+
+	var result vectorResult
+	require.NoError(t, json.Unmarshal(recorder.Body.Bytes(), &result))
+	require.Equal(t, 2, len(result.Data.Result))
+
+	at0, value0, err := result.Data.Result[0].Value.parse()
+	require.NoError(t, err)
+	at1, value1, err := result.Data.Result[1].Value.parse()
+	require.NoError(t, err)
+
+	expected := mustPrettyJSON(t, fmt.Sprintf(`
```

Comment:
nit: Think it may be a little cleaner to flip this around; have expected be a `vectorResult` and then just check equality between it and default json decoded result?

Reply:
Sure yeah.